test/functional/TypeDefinition.hs

```diff
@@ -1,102 +1,53 @@
-            defs <- getTypeDefinitions doc (toPos (11, 23))
-            liftIO $ do
-                fp <- canonicalizePath "test/testdata/gototest/src/Lib.hs"
-                defs @?= [ Location (filePathToUri fp)
-                                    (Range (toPos (8, 1)) (toPos (8, 29)))
-                         ]
-    , ignoreTestBecause "Broken" $ testCase "finds local definition of newtype variable"
-        $ runSession hlsCommand fullCaps "test/testdata/gototest"
-        $ do
-            doc  <- openDoc "src/Lib.hs" "haskell"
-            defs <- getTypeDefinitions doc (toPos (16, 21))
-            liftIO $ do
-                fp <- canonicalizePath "test/testdata/gototest/src/Lib.hs"
-                defs @?= [ Location (filePathToUri fp)
-                                    (Range (toPos (13, 1)) (toPos (13, 30)))
-                         ]
-    , ignoreTestBecause "Broken" $ testCase "finds local definition of sum type variable"
-        $ runSession hlsCommand fullCaps "test/testdata/gototest"
-        $ do
-            doc  <- openDoc "src/Lib.hs" "haskell"
-            defs <- getTypeDefinitions doc (toPos (21, 13))
-            liftIO $ do
-                fp <- canonicalizePath "test/testdata/gototest/src/Lib.hs"
-                defs @?= [ Location (filePathToUri fp)
-                                    (Range (toPos (18, 1)) (toPos (18, 26)))
-                         ]
-    , ignoreTestBecause "Broken" $ testCase "finds local definition of sum type contructor"
-            $ runSession hlsCommand fullCaps "test/testdata/gototest"
-            $ do
-                doc  <- openDoc "src/Lib.hs" "haskell"
-                defs <- getTypeDefinitions doc (toPos (24, 7))
-                liftIO $ do
-                    fp <- canonicalizePath "test/testdata/gototest/src/Lib.hs"
-                    defs
-                        @?= [ Location (filePathToUri fp)
-                                       (Range (toPos (18, 1)) (toPos (18, 26)))
-                            ]
-    , ignoreTestBecause "Broken" $ testCase "can not find non-local definition of type def"
-        $ runSession hlsCommand fullCaps "test/testdata/gototest"
-        $ do
-            doc  <- openDoc "src/Lib.hs" "haskell"
-            defs <- getTypeDefinitions doc (toPos (30, 17))
-            liftIO $ defs @?= []
+    testCase "finds local definition of record variable"
+        $ getTypeDefinitionTest' (11, 23) 8
+    , testCase "finds local definition of newtype variable"
+        $ getTypeDefinitionTest' (16, 21) 13
+    , testCase "finds local definition of sum type variable"
+        $ getTypeDefinitionTest' (21, 13) 18
+    , testCase "finds local definition of sum type constructor"
+        $ getTypeDefinitionTest' (24, 7) 18
+    , testCase "finds non-local definition of type def"
+        $ getTypeDefinitionTest' (30, 17) 27
+    , testCase "find local definition of type def"
+        $ getTypeDefinitionTest' (35, 16) 32
```

Comment:
Note: In the original test, the symbol who's type's definition we seek was at (35,16) and the type found was the data declaration for Enu at line 18 (data Enu = First | Second). Now we instead find a type definition for TypEnu at line 32 (type TypEnu = Enu).

The behavior of "Go to Type Definition" in this case:
```
data Enu = First | Second
type TypEnu = Enu
typEnuId :: TypEnu -> TypEnu
typEnuId enu = en[CURSOR HERE]u
```
seems to have changed. Formerly, we got the underlying type, now we get the type synonym. This is good though. Because in this case:

```
type MyInt = Int
myIntId :: MyInt -> MyInt
myIntId myInt = my[CURSOR HERE]Int
```
The new behavior will show us that MyInt is just an Int, while the old behavior would produce "No type definition found for myInt".